src/compiler/emitter.ts

```diff
@@ -859,15 +865,19 @@ export function emitFiles(resolver: EmitResolver, host: EmitHost, targetSourceFi
         const filesForEmit = forceDtsEmit ? sourceFiles : filter(sourceFiles, isSourceFileNotJson);
         // Setup and perform the transformation to retrieve declarations from the input files
         const inputListOrBundle = compilerOptions.outFile ? [factory.createBundle(filesForEmit)] : filesForEmit;
-        if (
-            (emitOnly && !getEmitDeclarations(compilerOptions)) ||
-            compilerOptions.noCheck ||
-            emitResolverSkipsTypeChecking(emitOnly, forceDtsEmit)
-        ) {
-            // Checker wont collect the linked aliases since thats only done when declaration is enabled and checking is performed.
-            // Do that here when emitting only dts files
-            filesForEmit.forEach(collectLinkedAliases);
-        }
+        // Checker wont collect the linked aliases since thats only done when declaration is enabled and checking is performed.
+        // Do that here when emitting only dts files
+        filesForEmit.forEach(sourceFile => {
+            if (
+                (emitOnly && !getEmitDeclarations(compilerOptions)) ||
+                compilerOptions.noCheck ||
+                emitResolverSkipsTypeChecking(emitOnly, forceDtsEmit) ||

```

Comment:
Aside, since it's not related to this PR, we probably wanna feed this to the JS emit-related conditions we just edited here if we ever intend for the `skipDiagnostics`/`forceDtsEmit` `EmitResolver` parameter to be usable with arbitrary JS emit modes, too. Don't think it matters right now, tho, since it's only used for .d.ts emit afaik.

Reply:
I didnt do this because dont you need to handle that at `calculateNodeCheckFlagWorker` and didnt want to store if calling from emitResolver etc there as there is no easy way to flow that information.